pkg/database/database.go

```diff
@@ -52,7 +53,18 @@ type Database struct {
 	Deleter  facade.DBDeleter
 }
 
+var db DBInterface
+var once sync.Once
+
 func NewDatabase() (DBInterface, error) {
+	var err error
+	once.Do(func() {
+		db, err = newDB()
+	})
+	return db, err
+}
+
+func newDB() (DBInterface, error) {
```

Comment:
Why not just embed the code for `newDB` in `newDatabase`?

Reply:
There is no special reason, I changed it to embed this piece of code in `newDatabase`